OsmApi.pm

```diff
@@ -143,6 +143,7 @@ sub put
     return dummylog("PUT", $url, $body) if ($prefs->{dryrun});
     my $req = HTTP::Request->new(PUT => $prefs->{apiurl}.$url);
     $req->header("Content-type" => "text/xml");
+    $req->header("Authorization" => "Basic ".encode_base64($prefs->{username}.":".$prefs->{password}));
```

Comment:
Are you sure about this? The ua->credentials line further up should take care of HTTP Basic Auth. The reason we have the Authorization header explicitly set in get_with_credentials is that some GET requests, while not rejected for un-authorized clients, will deliver different responses depending on whether you are authorized or not and this does not trigger the libwww credential-setting automatism. But for POST/PUT/DELETE, the reply is always "unauthorized" and then libwww sets the header automatically.